qiskit/synthesis/multi_controlled/mcx_synthesis.py

```diff
@@ -207,32 +209,19 @@ def synth_mcx_1_clean_b95(num_ctrl_qubits: int) -> QuantumCircuit:
     q_ancilla = q[-1]
     q_target = q[-2]
     middle = ceil(num_ctrl_qubits / 2)
-    first_half = [*q[:middle]]
-    second_half = [*q[middle : num_ctrl_qubits - 1], q_ancilla]
-
-    qc_first_half = synth_mcx_n_dirty_i15(num_ctrl_qubits=len(first_half))
-    qc_second_half = synth_mcx_n_dirty_i15(num_ctrl_qubits=len(second_half))
-
-    qc.append(
-        qc_first_half,
-        qargs=[*first_half, q_ancilla, *q[middle : middle + len(first_half) - 2]],
-        cargs=[],
-    )
-    qc.append(
-        qc_second_half,
-        qargs=[*second_half, q_target, *q[: len(second_half) - 2]],
-        cargs=[],
-    )
-    qc.append(
-        qc_first_half,
-        qargs=[*first_half, q_ancilla, *q[middle : middle + len(first_half) - 2]],
-        cargs=[],
-    )
-    qc.append(
-        qc_second_half,
-        qargs=[*second_half, q_target, *q[: len(second_half) - 2]],
-        cargs=[],
-    )
+
+    controls1 = [*q[:middle]]
+    mcx1 = synth_mcx_n_dirty_i15(num_ctrl_qubits=len(controls1), relative_phase=True)
```

Comment:
can you briefly explain why one can do the optimization of relative_phase=True here?

Reply:
done